### PR TITLE
Milestone 2: Add reducers and action

### PR DESCRIPTION
**Pull request by BenMKT**
# Project Implementations:

- Add Redux Toolkit.
- Created a directory that will contain all the Redux logic.
- Configured a Redux store.
- Defined a slice of state for books.
- Defined a slice of state for categories.
- No changes to the React components in this project yet.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: [],
+  reducers: {
+    checkStatus: () => {
+      const initialState = 'Under construction';
+      return initialState;
+    },
```
Comment (AbelG101) [inline src/redux/categories/categoriesSlice.js]: - [x] Ouch! this is a close one but as part of the project requirements you should check the status from the action payload before returning the message in the reducer, This means the reducer should accept `state` and `action` as parameters as shown below 👇.
```jsx
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @BenMKT 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Proper use of Redux ✔
- Correct implementation of Reducers and Slices ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (BenMKT) [reply src/redux/categories/categoriesSlice.js]: - Refactored to check the status from the action payload before returning the message in the reducer.

**Review by Olamarx — APPROVED** [review]
Hi @BenMKT,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Required changes fixed ✔️
- Good commit message ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**